### PR TITLE
render avg ratings on home page

### PR DESCRIPTION
**Pull request by TheunisK**

### CODE REVIEW

File: src/client/components/homeComponents/HomeRatings.jsx [hunk 1]
```diff
@@ -1,14 +1,118 @@
-import pikachuImg from "../../assets/pikachuTest.png";
+import { useState, useEffect } from 'react';
+
+const URL = process.env.REACT_APP_API_URL;
+const pokemonRatingsEndpoint = "/pokemon/ratings";
+const pokemonRatingsURL = URL + pokemonRatingsEndpoint;
```
Comment (Can-Tina) [inline src/client/components/homeComponents/HomeRatings.jsx:5]: Never hard coding again!

File: src/client/components/homeComponents/HomeRatings.jsx [hunk 2]
```diff
@@ -1,14 +1,118 @@
...
+                <img src={sortedByDesign[0].art} alt={sortedByDesign[0].name} />
+            }
+
             <h2>Companionship</h2>
-            <img src={pikachuImg} alt="Pikachu" />
+
+            {sortedByDesign.length &&
+                <img src={sortedByCompanion[0].art} alt={sortedByCompanion[0].name} />
+            }
+
```
Comment (Can-Tina) [inline src/client/components/homeComponents/HomeRatings.jsx:115]: This has come a long way :)

File: src/server/controllers/pokemon.js [hunk 3]
```diff
@@ -49,7 +49,7 @@ const createPokemons = async (req, res) => {
     }
     res.json("No Access")
 }
-
+/* This name needs changing and 1 is a magic number */
```
Comment (Can-Tina) [inline src/server/controllers/pokemon.js:52]: Is it a magic number?

Comment (nathanSamosa) [reply src/server/controllers/pokemon.js:52]: This comment refers to `getPokemons`, which is a function used to fetch the pokemon ratings of a specific user. The name needs changing into something more explicit, and `profileId: 1` needs updated to a request body/param.

Comment (Can-Tina) [reply src/server/controllers/pokemon.js:52]: I see. Yes that makes sense now!

File: src/client/components/homeComponents/HomeRatings.jsx [hunk 4]
```diff
@@ -1,14 +1,118 @@
...
 
 const HomeRatings = () => {
+    const [rawData, setRawData] = useState([]);
+    const [avgRatingsArr, setAvgRatingsArr] = useState([]);
+
+    /* setting states of sorted arrays by rating type for all pokemon */
+
+    let [sortedByDesign, setSortedByDesign] = useState([]);
+    let [sortedByStrength, setSortedByStrength] = useState([]);
+    let [sortedByCompanion, setSortedByCompanion] = useState([]);
```
Comment (vherus) [inline src/client/components/homeComponents/HomeRatings.jsx:16]: Why are you using `let` here?

File: src/client/components/homeComponents/HomeRatings.jsx [hunk 5]
```diff
@@ -1,14 +1,118 @@
...
+
+    /* function to calculate the average rating by type used in returnAvgRatings function */
+
+    const calcAvg = (type, ratingArr) => {
+        let ratingSum = 0;
+        for (let i = 0; i < ratingArr.length; i++) {
+            ratingSum = ratingSum + ratingArr[i][type];
+        }
+        let ratingAvg = ratingSum / ratingArr.length;
```
Comment (vherus) [inline src/client/components/homeComponents/HomeRatings.jsx:25]: Be careful with your use of `let`'s, `ratingAvg` never gets changed after being declared so it should be a const

File: src/client/components/homeComponents/HomeRatings.jsx [hunk 6]
```diff
@@ -1,14 +1,118 @@
...
+    let [sortedByStrength, setSortedByStrength] = useState([]);
+    let [sortedByCompanion, setSortedByCompanion] = useState([]);
+
+    /* function to calculate the average rating by type used in returnAvgRatings function */
+
+    const calcAvg = (type, ratingArr) => {
+        let ratingSum = 0;
+        for (let i = 0; i < ratingArr.length; i++) {
+            ratingSum = ratingSum + ratingArr[i][type];
+        }
```
Comment (vherus) [inline src/client/components/homeComponents/HomeRatings.jsx:24]: Nice loop. You may also want to try solving this same problem with `Array.prototype.reduce` to increase your ability further: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce

File: src/server/controllers/pokemon.js [hunk 7]
```diff
@@ -64,8 +64,18 @@ const getPokemons = async(req, res) => {
     res.json({ data: pokemons })
 }
 
+const getPokemonRatings = async (req, res) => {
+    const pokeRatings = await prisma.pokemon.findMany({
+        include: {
+            rating: true
+        }
+    })
+    res.json({pokemonRatingData: pokeRatings})
```
Comment (vherus) [inline src/server/controllers/pokemon.js:73]: Shouldn't this conform to the same format that the other routes in your API have? `{ data: pokeRatings }`

**Review by vherus — APPROVED** [review]
Nice work. I'd have liked to see this broken up into at least a couple more commits rather than all chunked into a single one - finding the balance is an ongoing process

**Pull request merged**